frontend/src/components/SearchBar/SearchBar.js

```diff
@@ -14,14 +14,24 @@ class SearchBar extends Component {
     this.handleSubmit = this.handleSubmit.bind(this);
   }
 
-  handleChange(event) {
-    this.setState({ searchTerm: event.target.value });
+  setRedirect = () => {
+    const { history } = this.props;
+    const { searchTerm } = this.state;
+    const queryString = `/?q=${searchTerm}`;
+    history.push(queryString, { searchTerm });
```

Comment:
Instead of passing this with location state, it might be preferable to just change the URL and pull the search term directly from the `this.props.location.search` on the Homepage (instead of `this.props.location.state`). That means that if a user types in the a URL with a search term, it'll work the way we expect.